### PR TITLE
feat(dcutr): add `ConnectionId` to upgrade success/failed events

### LINKED ISSUE #4519: `dcutr::Event::DirectConnectionUpgradeSucceeded` should include the `ConnectionId`
This would be useful to know, which of our connections is the direct connection to this peer.

### PR DESCRIPTION
**Pull request by dariusc93**
## Description

Resolves #4519.

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: protocols/dcutr/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
+## 0.11.0 - unreleased
+
+- Add ConnectionId to Event::DirectConnectionUpgradeSucceeded and Event::DirectConnectionUpgradeFailed
```
Comment (thomaseizinger) [inline protocols/dcutr/CHANGELOG.md]:
```suggestion
- Add `ConnectionId` to `Event::DirectConnectionUpgradeSucceeded` and `Event::DirectConnectionUpgradeFailed`.
```

File: protocols/dcutr/src/behaviour_impl.rs [hunk 2]
```diff
@@ -263,6 +266,7 @@ impl NetworkBehaviour for Behaviour {
             self.queued_events.extend([ToSwarm::GenerateEvent(
                 Event::DirectConnectionUpgradeSucceeded {
                     remote_peer_id: peer,
+                    connection_id: relayed_connection_id,
```
Comment (thomaseizinger) [inline protocols/dcutr/src/behaviour_impl.rs]: This is not correct I think! :)

`connection_id` should be the direct one.

Comment (thomaseizinger) [reply protocols/dcutr/src/behaviour_impl.rs]: Same with the other below.

**Review by thomaseizinger — COMMENTED** [review]
Thanks!

Can we expend the test to assert that this is definitely the direct connection? We should be able to do that by recording the address from `SwarmEvent::ConnetionEstablished` and checking that it is not a relayed address.

File: protocols/dcutr/src/behaviour_impl.rs (hunk 2 above)
Comment (dariusc93) [reply protocols/dcutr/src/behaviour_impl.rs]: Nice catch!

**Review by mxinden — COMMENTED** [review]
Missing test. Otherwise looks good to me. Thanks.

**Review by thomaseizinger — APPROVED** [review]

**Pull request merged**